Library/Homebrew/os/mac/universal_socket.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require "extend/ENV"
+require "socket"
+
+class UniversalSocket
+  def self.open
+    socket = UNIXSocket.open(ENV["HOMEBREW_ERROR_PIPE"], &:recv_io)
```

Comment:
Given this is identical on macOS and Linux it should just live in `Library/Homebrew/universal_server.rb` (with a better name like just `Socket`)